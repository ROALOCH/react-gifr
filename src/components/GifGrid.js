import React from 'react';
import PropTypes from 'prop-types';

import GifGridItem from './GifGridItem';
import useFetchGifs from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);

	return (
		<>
			<h3 className='gif-title'>{category}</h3>
			{loading && (
				<div id='loadingAlert' className='alert alert-secondary' role='alert'>
					Cargando Im√°genes
				</div>
			)}
			<div className='card-grid'>
				{images.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};

export default GifGrid;
