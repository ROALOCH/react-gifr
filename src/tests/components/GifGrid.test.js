import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>', () => {
	const category = 'Ozark';

	test('Debe mostrar el componente correctamente', () => {
		useFetchGifs.mockReturnValue({ data: [], loading: true });
		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs()', () => {
		const gifs = [
			{
				id: 'ABC',
				url: 'https://localhost/images/gif.jpg',
				title: 'Gif',
			},
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('#loadingAlert').exists()).toBe(false);
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
