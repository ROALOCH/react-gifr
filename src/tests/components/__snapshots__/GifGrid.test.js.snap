// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe de mostrar items cuando se cargan imágenes con useFetchGifs() 1`] = `
<Fragment>
  <h3
    className="gif-title"
  >
    Ozark
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Gif"
      url="https://localhost/images/gif.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid/> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="gif-title"
  >
    Ozark
  </h3>
  <div
    className="alert alert-info"
    id="loadingAlert"
    role="alert"
  >
    Cargando Imágenes
  </div>
  <div
    className="card-grid"
  />
</Fragment>
`;
